File: InorderTraversal.java
ArrayList<Integer> inOrder(Node root) {
        // Code
        ArrayList<Integer> res = new ArrayList<Integer>();
        inorderUtils(root, res);
        return res;
    }
    
    void inorderUtils(Node root, ArrayList<Integer> res){
        if(root == null){
            return;
        }
        
        inorderUtils(root.left, res);
        res.add(root.data);
        inorderUtils(root.right, res);
    }

File: InorderTraversal.cpp
vector<int> inOrder(Node* root) {
        // Your code here
        vector<int> res;
        inorderUtils(root, res);
        return res;
    }
    
    void inorderUtils(Node* root, vector<int>& res){
        if(root == nullptr){
            return;
        }
        
        inorderUtils(root->left, res);
        res.push_back(root->data);
        inorderUtils(root->right, res);
    }
